[mariia] iP

popova-mariia opened the pull request:
**MariiaChatBot**
>“Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

MariiaChatBot frees your mind of having to remember things you need to do. It's,

* text-based
* easy to learn
* ~~FAST~~ SUPER FAST to use
* All you need to do is,

1. download it from [here](https://github.com/popova-mariia/ip.git).
2. let it manage your tasks for you 😉
And it is **FREE!**

Features:

 - [x] Managing tasks
 - [ ] Managing deadlines (coming soon)
 - [x] Reminders (coming soon)
 
---
 
 If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```
    public static void main(String[] args) {
        new MariiaChatbot("./src/main/java/data/mariia.txt").run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

popova-mariia committed:
added simple greeting

popova-mariia committed:
able to add tasks

popova-mariia committed:
able to mark tasts as done

popova-mariia committed:
added different types of tasks

popova-mariia committed:
testing

popova-mariia committed:
added error handling

popova-mariia committed:
added delete option adn updated the testing

popova-mariia committed:
Add data folder with mariia.txt under src/main/java

Xavierlhm commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,18 @@
+    }
+
+    @Override
+    public String getType() {
```
Good job following the coding standard for the method naming! 👍

Xavierlhm commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,20 @@
+    }
+
+    @Override
+    public String getType() {
```
Good job following the coding standard for the method naming! 👍

Xavierlhm commented on src/main/java/MariiaChatbot.java:
```diff
@@ -0,0 +1,104 @@
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.util.List;
```
Good job following the coding standard for the importing of classes! 👍

Xavierlhm commented on src/main/java/MariiaChatbot.java:
```diff
@@ -0,0 +1,104 @@
+                    System.out.println(" Bye. Hope to never see you again!");
+                    break;
+                }
+                else if (input.equals("list")) {
```
Should this line of ```else if``` statement be on line 22 instead?

Xavierlhm commented on src/main/java/MariiaChatbot.java:
```diff
@@ -0,0 +1,104 @@
+                        System.out.println((i + 1) + "." + list.get(i));
+                    }
+
+                }else if (input.startsWith("delete")) {
```
Perhaps a spacing after the curly bracket?

Xavierlhm commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,24 @@
+public abstract class Task {
+    String description;
+    boolean isDone;
```
Good job following the coding standard for the naming of Boolean variables! 👍

Xavierlhm reviewed (commented):
Good job following the coding standard! Just a few touch ups and its good to go! 😄

popova-mariia committed:
Commit old edits

popova-mariia committed:
Add HardDisk class.
Format to fit Java coding standard.

popova-mariia committed:
Remove .class files from repository

popova-mariia committed:
Merge branch-Level-7 into master

popova-mariia committed:
Add dates and times

popova-mariia committed:
Add dates and times.
Dates and times are now printed in yyyy-mm-dd format.
Fix quitting the program every time an error occurs due to wrong input: make program continue to let user correct input.

popova-mariia committed:
Merge branch-Level-8 into master

popova-mariia committed:
Merge branch-Level-8 to master branch

popova-mariia committed:
Add more OOP principles

popova-mariia committed:
Merge A-MoreOOP to master branch

popova-mariia committed:
Add packages

popova-mariia committed:
Merge branch 'add-gradle-support' to master branch

SahanaAvantika commented on src/main/java/mariiachatbot/storage/HardDisk.java:
```diff
@@ -0,0 +1,182 @@
+     * @return A task which was represented by a string in the file.
+     * @throws EmptyDescriptionException If some part of the command (ie description, by, from and to) is absent.
+     */
+    public Task parseTask(String str) {
```
Nice! You used verbs for your functions

popova-mariia committed:
Edit according to java coding standards with checkstyle

popova-mariia committed:
Add JUnit testing for HardDisk and Parser classes

popova-mariia committed:
Edit JavaDoc

popova-mariia committed:
Add a comment

popova-mariia committed:
Add find option

popova-mariia committed:
Merge branch 'branch-A-JavaDoc' into master branch.

popova-mariia committed:
Merge branch 'branch-A-CodingStandard' into master branch.

popova-mariia committed:
Merge branch 'branch-Level-9' into master branch.

popova-mariia committed:
Add automated checkstyle

popova-mariia committed:
Merge branch 'branch-A-CheckStyle' into master branch

popova-mariia committed:
Add Main, Launcher, DialogBox and MainWindow classes

popova-mariia committed:
Refine UI layout and styling

popova-mariia committed:
Add echoing the user

popova-mariia committed:
Add dialog boxes for the chatbot response

popova-mariia committed:
Add custom behavior to DialogBox

popova-mariia committed:
Add FXML

popova-mariia committed:
Merge branch 'branch-Level-10'into master branch

popova-mariia committed:
Complete GUI for MariiaChatBot

popova-mariia committed:
Add GUI command execution and UI responses

The program currently relies on a terminal-based UI, which is limiting.

A GUI is more user-friendly and can provide additional features.

Let's,
* Create Main to set up the chatbot
* Add MainWindow to display user input and bot output
* Establish communication between MainWindow and MariiaChatBot
 so that commands are executed properly and their output is reflected

popova-mariia committed:
Add assert feature

popova-mariia committed:
Change naming in main to ensure consistency

popova-mariia committed:
Merge pull request #1 from popova-mariia/branch-A-Assertions

[mariia] Add assertions

popova-mariia committed:
Merge pull request #2 from popova-mariia/branch-A-CodeQuality

[mariia] Edit to ensure code quality

popova-mariia committed:
Implement UndoCommand to revert last modifying action

The chatbot currently doesnt have  undo feature.

If a user makes a mistake, he/she must manually restore the
previous state, which is inconvenient.

With undo feature, user will save time on manually
undoing last action.

Let's,
* Track the most recent modifying action in `latest-changes.txt`.
* Handle undoing task creation (`todo`, `deadline`, `event`) by
  removing  latest task.
* Restore deleted tasks to their original positions instead of
  appending them.
* Revert `mark` and `unmark` actions by toggling the task’s state.

popova-mariia committed:
Merge tag 'branch-BCD-Extension'

popova-mariia committed:
Update chatbot: rename, change phrases, set font to yellow, update avatars

popova-mariia committed:
Refactor Parser: Extract certain command parsing into separate methods

popova-mariia committed:
Add error handling for unexpected date inputs

popova-mariia committed:
Add automated JUnit testing for Ui

popova-mariia committed:
Add a small change in spacing of ui messages

popova-mariia committed:
Edit README.md file to add user guide

popova-mariia committed:
Fix automated testing

popova-mariia committed:
Edit README.md Testing Johnny section

popova-mariia committed:
Edit JavaDoc comment for Ui.java

popova-mariia committed:
Add styles.css

popova-mariia committed:
Add background image

popova-mariia committed:
Change send button and text field

popova-mariia committed:
Change background image to final version

popova-mariia committed:
Add final changes

popova-mariia committed:
Fix creating new folder data

popova-mariia committed:
Add image to README.md

popova-mariia committed:
Correct table formatting in README.md
